### DISCUSSION TITLE
Thoughts on permlinks?

### DISCUSSION THREAD

**Discussion by andrewlow** [photoprism/photoprism#977] [Ideas]
Edit: seems there is #466 to capture this already.

While my primary use of photoprism is hosting it locally with no remote access - thus if you're not on my local network - you can't see it.

I still would like the ability to share
- search queries via links
- specific photo views via links

the UX I'm imagining is that for search - I'd compose my search query, and the URL in would change to reflect the query. Then by copy & pasting that URL from my browser - I'd be able to save or share the search.

Similarly, if I'm viewing a photo - I could copy & paste the URL the browser was pointed at - and that link would allow me to see exactly that photo.

This is intended to spark discussion on if this is a good idea - and if there is general agreement, maybe we can get it into the backlog for future releases.  Apologies if this is already on the roadmap or has been discussed

**Comment by alexisLefebvre** [comment]
So, you would appreciate to have a short link instead of something like https://demo.photoprism.org/browse?view=cards&color=green&order=added&q=berlin&public=true&quality=3?

**Reply by andrewlow** [reply]
Doh.. yeah - search is already correct 

I was looking at / thinking about the single photo view.  I'd like to point at a given photo no matter how I've managed to locate it via the UI

![image](https://user-images.githubusercontent.com/2952475/106393893-8cfd8a00-63c7-11eb-8250-2302b52ac3ff.png)

**Reply by alexisLefebvre** [reply]
This is not as easy as a permalink but you can do it with the `name` filter: https://demo.photoprism.org/browse?view=cards&color=green&order=added&q=name%3A%2220200831-180054-Bench-Berlin-2020-2wy.jpg%22&quality=3

**Reply by andrewlow** [reply]
That's closer - but requires some search magic.

I'm fine with the search permlink story - but overall - it should be easier to have permlinks to almost every view you have in the UI. 

Is the agreement on that? Or is search the only permlink story long term?

**Reply by alexisLefebvre** [reply]
I don't know about sharing search results but about sharing a single file, an issue exists: #466

**Reply by lastzero** [reply]
Obviously we'll add links to single images, also to be able to share them properly - right now, you may only share the raw image preview URL (stays the same to avoid cache busting unless the preview URL token was changed).

We'll take care of all issues long term, we just can't deliver new features every day - please be patient or help us with funding so that we can pay additional full-time developers :wink:

**Reply by andrewlow** [reply]
To be honest - I wasn't aware of #466 - and I did search for an issue before raising this discussion thread.

I also write software and have in the past maintained open source - it is a mostly thankless job. Thank you all for what you're doing here. I think #466 is probably close enough to the right answer - and hopefully others looking for permlinks will find this thread.
